src/PIL/BmpImagePlugin.py

```diff
@@ -328,6 +330,69 @@ def decode(self, buffer):
         return -1, 0
 
 
+class BmpRle4Decoder(ImageFile.PyDecoder):
+    _pulls_fd = True
+
+    def decode(self, buffer):
+        data = bytearray()
+        x = 0
+        while len(data) < self.state.xsize * self.state.ysize:
+            pixels = self.fd.read(1)
+            byte = self.fd.read(1)
+            if not pixels or not byte:
+                break
+            num_pixels = pixels[0]
+            if num_pixels:
+                # encoded mode
+                if x + num_pixels > self.state.xsize:
+                    # Too much data for row
+                    num_pixels = max(0, self.state.xsize - x)
+                first_pixel = o8(byte[0] >> 4)
+                second_pixel = o8(byte[0] & 0x0F)
+                for index in range(num_pixels):
+                    if index % 2 == 0:
+                        data += first_pixel
+                    else:
+                        data += second_pixel
+                x += num_pixels
+            else:
+                if byte[0] == 0:
+                    # end of line
+                    while len(data) % self.state.xsize != 0:
+                        data += b"\x00"
+                    x = 0
+                elif byte[0] == 1:
+                    # end of bitmap
+                    break
+                elif byte[0] == 2:
+                    # delta
+                    bytes_read = self.fd.read(2)
+                    if len(bytes_read) < 2:
+                        break
+                    right, up = self.fd.read(2)
+                    data += b"\x00" * (right + up * self.state.xsize)
+                    x = len(data) % self.state.xsize
+                else:
+                    # absolute mode (2 pixels per byte)
+                    total_bytes_to_read = byte[0] // 2
```

Comment:
What does this "2" comment mean? If it means this value is always 2, that won't work, because this is already checked for on line 367.

Reply:
It's not a comment. Commenting in Python is #. This is floor division.

Reply:
ah, right. There's even a real comment right above that and I didn't notice...